### ISSUE TITLE
build-env isn't finding easy_install.pth files for python egg packages

### ISSUE THREAD

**Issue by bdunlay** [LLNL/spack#27076]
### Steps to reproduce

* Create a package with a build-time dependency on python and py-pyquaternion
* Install the package
* Enter build-env and try to import

```
# spack install <spec>
# spack build-env <spec> -- bash
# which python
/opt/spack/software/linux-ubuntu18.04-x86_64/gcc-9.2.0/python-3.8.6-tpw6u7xosborjvrb35nzpu4m2zvebl7e/bin/python
root@u78cb42cd935758:/opt/env# python
Python 3.8.6 (default, Oct 28 2021, 19:29:38)
[GCC 9.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyquaternion
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'pyquaternion'
```

This looks to be the same problem as issue https://github.com/spack/spack/issues/6131, but different in that build-env is active without a view. The package can be imported in the **runtime** view.

For me, the consequence of this bug is that spack install --test=root fails.

### Information on your system

# spack debug report
* **Spack:** 0.16.1
* **Python:** 3.6.9
* **Platform:** linux-ubuntu18.04-skylake

Example dependency declaration in package.yaml
```
    depends_on('python@3.8:', type=('build', 'test', 'run'))
    depends_on('py-pyquaternion', type=('build', 'test', 'run'))
```

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by adamjstewart** [comment]
Is this fixed by #28346?

**Comment by alalazo** [comment]
Closing, since there has been no activity in a while. Feel free to reopen if you want to continue work on this.
